clang_bind/cmake_frontend.py

```diff
@@ -0,0 +1,162 @@
+
+    def get_compile_groups(self):
+        return [
+            {
+                "fragments": [
+                    compile_command_fragment.get("fragment")
+                    for compile_command_fragment in compile_group.get(
+                        "compileCommandFragments", []
+                    )
+                ],
+                "defines": [
+                    define.get("define") for define in compile_group.get("defines", [])
+                ],
+                "includes": [
+                    include.get("path") for include in compile_group.get("includes", [])
+                ],
+                "language": compile_group.get("language"),
+            }
+            for compile_group in self.get_target().get("compileGroups", [])
+        ]
+
+    def get_dependencies(self):
+        return [
+            dependency.get("id")
+            for dependency in self.get_target().get("dependencies", [])
+        ]
+
+    def get_files(self):
+        return self.get_target().get("backtraceGraph", {}).get("files")
+
+    def get_folder(self):
+        return self.get_target().get("folder", {}).get("name")
+
+    def get_id(self):
+        return self.get_target().get("id")
+
+    def get_install(self):
+        install = self.get_target().get("install", {})
+        return {
+            "destinations": [
+                destination.get("path")
+                for destination in install.get("destinations", [])
+            ],
+            "prefix": install.get("prefix", {}).get("path"),
+        }
+
+    def get_link(self):
+        link = self.get_target().get("link", {})
+        command_fragments = link.get("commandFragments", [])
+        return {
+            "flags_fragments": [
+                command_fragment.get("fragment")
+                for command_fragment in command_fragments
+                if command_fragment.get("role") == "flags"
+            ],
+            "libraries_fragments": [
+                command_fragment.get("fragment")
+                for command_fragment in command_fragments
+                if command_fragment.get("role") == "libraries"
+            ],
+            "language": link.get("language"),
+        }
+
+    def get_name(self):
+        return self.get_target().get("name")
+
+    def get_name_on_disk(self):
+        return self.get_target().get("nameOnDisk")
+
+    def get_paths(self):
+        return self.get_target().get("paths")
+
+    def get_sources(self):
+        return [sources.get("path") for sources in self.get_target().get("sources", [])]
+
+    def get_type(self):
+        return self.get_target().get("type")
+
+
+class CMakeFileAPI:
+    def __init__(self, build_dir, targets=None):
+        reply_dir = os.path.join(build_dir, ".cmake", "api", "v1", "reply")
+        if targets:
+            self.targets = [Target(reply_dir, target) for target in targets]
+        else:
+            self.targets = [
+                Target(reply_dir, codemodel_target)
```

Comment:
What's codemodel_target?

Reply:
Changed, not included in the new implementation.